src/v/storage/snapshot.cc

```diff
@@ -75,8 +77,8 @@ ss::future<> snapshot_manager::finish_snapshot(snapshot_writer& writer) {
 }
 
 ss::future<> snapshot_manager::remove_partial_snapshots() {
-    std::regex re(
-      fmt::format("^{}\\.partial\\.(\\d+)\\.([a-zA-Z0-9]{{4}})$", _filename));
+    std::regex re(fmt::format(
+      "^{}\\.partial\\.(\\d+)\\.([a-zA-Z0-9]{{4}})\\.(\\d+)$", _filename));
```

Comment:
does this regex mean that removing partial snapshots won't work for the previous naming format? that needs to continue to work to support removing partial snapshots that are left on disk after a crash

Reply:
Good idea I didn't think about that